test_libs/pyspec/eth2spec/utils/ssz/ssz_typing.py

```diff
@@ -1,142 +1,171 @@
+from typing import Dict, Iterator
 from types import GeneratorType
-from typing import List, Iterable, TypeVar, Type, NewType
-from typing import Union
-from typing_inspect import get_origin
 
-# SSZ integers
-# -----------------------------
 
+class DefaultingTypeMeta(type):
+    def default(cls):
+        raise Exception("Not implemented")
 
-class uint(int):
+
+class SSZType(DefaultingTypeMeta):
+
+    def is_fixed_size(cls):
+        raise Exception("Not implemented")
+
+
+class SSZValue(object, metaclass=SSZType):
+
+    def type(self):
+        return self.__class__
+
+
+class BasicType(SSZType):
     byte_len = 0
 
+    def is_fixed_size(cls):
+        return True
+
+
+class BasicValue(int, SSZValue, metaclass=BasicType):
+    pass
+
+
+class Bit(BasicValue):  # can't subclass bool.
```

Comment:
In https://github.com/ethereum/eth2.0-specs/blob/dev/specs/simple-serialize.md#basic-types, it's still be known as `bool`. I think either (i) adding some description of `Bit` in the phase 0 spec, or (ii) updating the markdown spec.

I guess (i) would be less overhead for the implementers.

Reply:
This was resolved a bit ago, it's `Bool` now, to keep it within the typing support of ssz. And we've `Bit` as alias (subclass of `Bool`).